### PR TITLE
Raise an exception when filters' value isn't a list.

### PR DESCRIPTION
**Pull request by arbaobao**
### Setup process

- Ran a remote.recent_executions() with filters using list or other types.
```
from flytekit.configuration import Config
from flytekit.models.filters import ValueIn, Contains
from flytekit.remote import FlyteRemote

remote = FlyteRemote(
    Config.auto(config_file="/home/nelson/.flyte/config-sandbox.yaml"),
    default_project="flytesnacks",
    default_domain="development",
)

executions = remote.recent_executions(
        project="flytesnacks",
        domain="development",
        limit=1,
        filters=[
            ValueIn("phase", "RUNNING"),
        ],
)
```

### Screenshots
image

image

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

File: flytekit/models/filters.py [hunk 1]
```diff
@@ -118,8 +118,11 @@ def __init__(self, key, values):
         :param Text key:  The name of the field to compare against
         :param list[Text] values:  A list of textual values to compare.
         """
+        if not isinstance(values, list):
+            raise TypeError("values must be a list.")
```
Comment (pingsutw) [inline flytekit/models/filters.py:122]:
```suggestion
            raise TypeError(f"values must be a list, but got {type(value)}")
```

**Comment by eapolinario** [comment]
Can you add a unit test?

**Comment by thomasjpfan** [comment]
@arbaobao This is not signed off yet. You can find instructions on how to sign off here: https://github.com/flyteorg/flytekit/pull/2576/checks?check_run_id=28277359939

**Comment by pingsutw** [comment]
@arbaobao there is a linting error, could you run `make lint` locally and push it again?

**Review by pingsutw — DISMISSED** [review]
lgtm

**Review by pingsutw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Pull request merged**